cmd/neofs-adm/internal/modules/morph/initialize.go

```diff
@@ -265,37 +255,44 @@ func (c *initializeContext) nnsContractState() (*state.Contract, error) {
 	return cs, nil
 }
 
-func (c *initializeContext) getSigner(tryGroup bool, acc *wallet.Account) transaction.Signer {
-	if tryGroup && c.groupKey != nil {
-		return transaction.Signer{
-			Account:       acc.Contract.ScriptHash(),
-			Scopes:        transaction.CustomGroups,
-			AllowedGroups: keys.PublicKeys{c.groupKey},
-		}
-	}
-
-	signer := transaction.Signer{
+func (c *initializeContext) getSigner(fancyScope bool, acc *wallet.Account) transaction.Signer {
+	var signer = &transaction.Signer{
 		Account: acc.Contract.ScriptHash(),
-		Scopes:  transaction.Global, // Scope is important, as we have nested call to container contract.
+		Scopes:  transaction.CalledByEntry,
 	}
 
-	if !tryGroup {
-		return signer
+	if !fancyScope {
+		return *signer
 	}
 
-	nnsCs, err := c.nnsContractState()
+	nnsHash, err := nns.InferHash(c.Client)
 	if err != nil {
-		return signer
+		return *signer
 	}
 
-	groupKey, err := nnsResolveKey(c.ReadOnlyInvoker, nnsCs.Hash, morphClient.NNSGroupKeyName)
-	if err == nil {
-		c.groupKey = groupKey
+	var nnsReader = nns.NewReader(invoker.New(c.Client, nil), nnsHash)
 
-		signer.Scopes = transaction.CustomGroups
-		signer.AllowedGroups = keys.PublicKeys{groupKey}
+	balanceHash, err := nnsReader.ResolveFSContract(nns.NameBalance)
+	if err != nil && c.Contracts[balanceContract] != nil {
```

Comment:
but what if `err != nil` and no hash has been cached?

Reply:
It means that we don't know the hash and it stays zero. In this case it's irrelevant, we don't have data from contract, we don't have data from the network, let's hope it works, but if it doesn't we can't help much.

Reply:
passing zero hash and continuing working scares me. but ok